### ISSUE TITLE
Check to see if library is header only from script

### ISSUE THREAD

**Issue by Lyberta** [conan-io/conan#3031]
I'm updating my CI script and I've stumbled on a problem that I don't know how to check if library I'm building and deploying is header only. For regular libraries I build several configurations but for header only I need only to build once.

**Comment by memsharded** [comment]
Hi @Lyberta 

I think a variable in your CI code with a list of header-only packages would be good enough. I am not sure what are you thinking about. Please note that even header-only packages might have settings, because they want to run CI that will run unit-tests in many different platforms, to make sure the code is OK, even if it is packaged only once. So this is not a task of the package manager, the CI should decide which configurations is going to build for which packages.

**Comment by Lyberta** [comment]
I'm thinking of a way to expose `self.info.header_only()` to CI script. In my case I have each package in a separate repo so having a list of header only libraries wouldn't scale.

Of course, I can place a separate file and make CI script check for existence of this file. But why duplicate something that is already inside `conanfile.py`?

**Comment by lasote** [comment]
I don't know if it helps but a header-only library that adjusts `self.info.header_only()` will always generate a package with the ID `5ab84d6acfe1f23c4fae0ab88f26e3a396351ac9`.  Maybe your CI could check it and stop? I don't know. Otherwise just search that there is no `settings=` string in the conanfile and  `self.info.header_only()`  string is there.

**Comment by memsharded** [comment]
Exposing ``self.info.header_only()`` is not a complete solution, as in fact, there can and there are header only packages that do not have such line. It is only necessary if the header-only package has transitive dependencies and/or is using settings to build and run tests.

**Comment by Lyberta** [comment]
Package ID trick worked wonderfully. Thanks.
